### PR TITLE
Redirect users when they cannot access content

### PR DESCRIPTION
**Pull request by abutterworth**
TNL-7171, TNL-7172, TNL-7173, TNL-7174: When a user is denied access to load courseware, redirect them to the appropriate location based upon the error code returned. If the error code is unknown they will be redirected to course home.

Depends on: https://github.com/edx/edx-platform/pull/23651

### CODE REVIEW

File: src/courseware/CoursewareContainer.jsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React, { useEffect, useCallback } from 'react';
 import PropTypes from 'prop-types';
 import { useSelector, useDispatch } from 'react-redux';
 import { history } from '@edx/frontend-platform';
-
+import { getLocale } from '@edx/frontend-platform/i18n';
```
Comment (davidjoy) [inline src/courseware/CoursewareContainer.jsx:5]: This is also available in `AppContext` I believe, FYI.

Comment (abutterworth) [reply src/courseware/CoursewareContainer.jsx:5]: Will switch

**Review by davidjoy — APPROVED** [review]

File: src/courseware/CoursewareContainer.jsx (hunk 1 above)
Comment (abutterworth) [reply src/courseware/CoursewareContainer.jsx:5]: Will not switch after discussing

**Comment by abutterworth** [comment]
Waiting for https://github.com/edx/edx-platform/pull/23651 before merge

**Pull request merged**